crates/nu-protocol/src/engine/engine_state.rs

```diff
@@ -1549,6 +1549,38 @@ impl<'a> StateWorkingSet<'a> {
         self.num_files() - 1
     }
 
+    fn find_non_whitespace_index(contents: &[u8], start: usize) -> usize {
+        let whitespace_chars: Vec<u8> = Vec::from([9, 10, 11, 12, 13, 32]);
+
+        let mut cur_pos = start;
+        while cur_pos < contents.len() {
+            if whitespace_chars.contains(&contents[cur_pos]) {
+                cur_pos += 1;
+            } else {
+                break;
+            }
+        }
+
+        cur_pos
+    }
+
+    pub fn is_sudo(&self) -> bool {
+        for (contents, _, _) in &self.delta.file_contents {
+            let last_pipe_pos = contents.iter().rev().position(|x| x == &124_u8);
```

Comment:
No magic numbers please.
```suggestion
            let last_pipe_pos = contents.iter().rev().position(|x| x == b'|');
```